src/rustup-cli/self_update.rs

```diff
@@ -1448,7 +1448,17 @@ pub fn prepare_update() -> Result<Option<PathBuf>> {
     }
 
     // Get build triple
-    let triple = dist::TargetTriple::from_build();
+    let triple = match dist::TargetTriple::from_build() {
+        // For windows x86 builds seem slow when used with windows defender.
+        // The website defaulted to i686-windows-gnu builds for a long time.
+        // This ensures that we update to a version thats appropriate for users
+        // and also works around if the website messed up the detection.
+        // If someone really wants to use another version, he still can enforce
+        // that using the environment variable RUSTUP_OVERRIDE_HOST_TRIPLE.
+        #[cfg(windows)]
```

Comment:
This ended up causing a build warning on windows, could `if cfg!(windows)` be used instead?